### PR TITLE
Issue #9272: Create parser utility class

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #9272: Create parser util class that will host util methods for parser that creates AST

### CODE REVIEW

**Review by pbludov — APPROVED** [review]

**Comment by nmancus1** [comment]
@pbludov team city failure appears to be a timeout, can you restart please?

File: config/import-control.xml [hunk 1]
```diff
@@ -59,9 +59,12 @@
...
+  <file name="ParserUtil">
+    <allow class="antlr.Token"/>
+    <allow class="com.puppycrawl.tools.checkstyle.DetailAstImpl"/>
+  </file>
+
   <subpackage name="utils">
-    <!-- until https://github.com/checkstyle/checkstyle/issues/9272 -->
-    <allow pkg="antlr"/>
     <!-- until https://github.com/checkstyle/checkstyle/issues/9271 -->
     <allow class="com.puppycrawl.tools.checkstyle.DetailAstImpl"/>
```
Comment (strkkk) [inline config/import-control.xml:69]: this line should be removed as well

Comment (nmancus1) [reply config/import-control.xml:69]: Shouldn't this be removed in #9271 ?

Comment (strkkk) [reply config/import-control.xml:69]: no, it was messed up a little. You can remove this line and reference to issue as well

Comment (romani) [reply config/import-control.xml:69]: Issue is about different line in this file.
@nmancus1 , please try to remove this allowance in until package.

Comment (nmancus1) [reply config/import-control.xml]: Done.

File: config/import-control.xml [hunk 2]
```diff
@@ -58,12 +58,13 @@
   <file name="PropertyCacheFile">
     <allow class="java.math.BigInteger"/>
   </file>
+  <file name="ParserUtil">
+    <allow class="antlr.Token"/>
+    <!-- commented out until https://github.com/checkstyle/checkstyle/issues/9271 -->
+    <!-- <allow class="com.puppycrawl.tools.checkstyle.DetailAstImpl"/> -->
```
Comment (nmancus1) [inline config/import-control.xml:64]: Added this here so that we do not forget in our work on #9271 to activate this.

Comment (romani) [reply config/import-control.xml:64]: Please remove.
We do not need code in advance, you can place comment to issue

**Review by romani — APPROVED** [review]
Item:

**Comment by nmancus1** [comment]
> We do not need code in advance, you can place comment to issue

Done at https://github.com/checkstyle/checkstyle/issues/9271#issuecomment-787392618

File: config/import-control.xml (hunk 2 above)
Comment (nmancus1) [reply config/import-control.xml]: Done.

**Review by romani — APPROVED** [review]
Ok to merge

**Review by strkkk — APPROVED** [review]

**Pull request merged**